### PR TITLE
docs: adding more documentation to the readme

### PR DESCRIPTION
**Pull request by basselalaraaj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
 # graphql-schema-registry
 [![Go Reference](https://pkg.go.dev/badge/github.com/basselalaraaj/graphql-schema-registry.svg)](https://pkg.go.dev/github.com/basselalaraaj/graphql-schema-registry)
 
+GraphQL Schema registry is used for discovery of services and it pushed new updates of the schema's automatically to the gateway, keeping the gateway always up to date.
```
Comment (thijsgadiot) [inline README.md]:
```suggestion
GraphQL Schema registry is used for discovery of services and keeps the schema of the gateway in sync with the service schemas it consumes, by automatically pushing schema changes.
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+
+When using a GraphQL gateway together with multiple GraphQL services, you will run into the issue that the gateway requires static paths to connect to multiple GraphQL services.
+Using introspection the gateway reads the schema's of the GraphQL services and creates its own schema. But if introspection of one of the GraphQL services fails for example due to timeout, the gateway will fail generating the schema, resulting in a failing gateway. By validating the schema's and caching them in a schema registry, you only depend on one call to fetch all the required schema's to build the gateway schema. Lowering the risk of having an unavailable gateway.
```
Comment (thijsgadiot) [inline README.md]: schema’s -> schemas

File: README.md [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+
+The gateway reflects the schema's of multiple GraphQL services. When a GraphQL service releases a new schema, the gateway should be notified with the schema changes to avoid new queries that are not available due to an outdated gateway. By pushing the GraphQL schema to the registry, after storing the schema, the registry pushes the new GraphQL schema to the gateway. This way all new operation in the underlying GraphQL services, are made available through the gateway.
```
Comment (thijsgadiot) [inline README.md]: Should the gateway be notified with the actual changes or only with the fact that there are changes (after which it does a fetch itself)?

File: server.go [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// GraphQL Schema registry is used for discovery of services and it pushed new updates of the schema's
```
Comment (thijsgadiot) [inline server.go]:
```suggestion
// GraphQL Schema registry is used for discovery of services and it pushes new updates of the schemas
```

File: README.md (hunk 3 above)
Comment (basselalaraaj) [reply README.md]: Both scenarios are possible, we send the schema as well in the message to the message broker. You can choose to ignore it and refetch.

**Pull request merged**